File: subscriber.js
var PB = require('pubnub');

pub_key = process.env.PUB_KEY;
sub_key = process.env.SUB_KEY;
sec_key = process.env.SEC_KEY;

var pubnub = PB({
	publish_key : pub_key,
	subscribe_key : sub_key,
	secret_key : sec_key,
	cipher_key : '',
	ssl_on : false,
});

var frequency_count = function (text) {
	var i, count;
	count = {};
	for (i = 0; i < text.length; i++) {
		count[text[i]] = (count[text[i]] || 0) + 1;
	}
	return count;
};

pubnub.subscribe({
	channel: "a",
	callback: function (message, channel) {
		var out = {};
		out.message = message;
		out.channel = channel;
		out.frequency_count = frequency_count(message);
		console.log(JSON.stringify(out));	
	},
	error: function (message) {
		console.log( "node received: " +  JSON.stringify({"error" : message}) );
	}
});

File: subscriber.py
from collections import defaultdict
import re
import json
import os
import Pubnub as PB

PUB_KEY = os.environ["PUB_KEY"]
SUB_KEY = os.environ["SUB_KEY"]
SEC_KEY = os.environ["SEC_KEY"]
CHANNEL_NAME = os.environ["CHANNEL_NAME"]

def frequency_count(text):
    "determine count of each letter"
    count = defaultdict(int)
    for char in text:
        count[char] += 1
    return count

def callback(message, channel):
    "print message, channel, and frequency count to STDOUT"
    print("python recevied:" + str({
        "channel": channel,
        "message": message,
        "frequency count":dict(frequency_count(message)),
    }))
    
def error(message):
    print({
        "error" : message
    })

if __name__ == "__main__":
    PB.Pubnub(
        publish_key = PUB_KEY,
        subscribe_key = SUB_KEY,
        secret_key = SEC_KEY,
        cipher_key = '',
        ssl_on = False,
    ).subscribe(
        channels=CHANNEL_NAME,
        callback=callback,
        error=error
    )
